libs/azure-iot-middleware-freertos/libraries/azure-sdk-for-c/sdk/inc/azure/iot/az_iot_adu_client.h

```diff
@@ -62,6 +77,7 @@ typedef struct
 {
   az_span manufacturer;
   az_span model;
+  az_iot_adu_device_custom_properties custom_properties;
```

Comment:
with the middleware, the device properties is a pointer which can be `NULL`'d. Should this also be a pointer which the user then allocates and passes themselves?

Reply:
discussed offline. this is due to the layering with the middleware which would break the encapsulation of that and this SDK.

Reply:
The issue I found is that if we define the property as a pointer in the embedded c sdk as well, now we need to create a struct anyway in the middleware that is always there, available to be filled up and provided to the embedded c sdk if the user so defines it.

Being discussed offline.